svsim/src/main/scala/Backend.scala

```diff
@@ -145,6 +145,9 @@ trait Backend {
     backendSpecificSettings: CompilationSettings
   ): Backend.Parameters
 
+  /** Get the waveform filename from backend-specific settings */
+  def getWaveformFilename(settings: CompilationSettings): Option[String]
```

Comment:
Nit: for any functions talking about "filenames" I'd be exact and refer to these as "stems" if they don't have a suffix or if they will be modified later to add a suffix.